src/path.jl

```diff
@@ -765,3 +765,21 @@ Base.mkpath(fp::AbstractPath) = mkdir(fp; recursive=true)
 # ALIASES for now old FilePaths API
 move(src::AbstractPath, dest::AbstractPath; kwargs...) = mv(src, dest; kwargs...)
 remove(fp::AbstractPath; kwargs...) = rm(fp; kwargs...)
+
+"""
+	isdescendent(child::AbstractPath, parent::AbstractPath) -> Bool
+
+Returns `true` if `child` is within the directory tree of `parent`.
+"""
+function isdescendent(child::AbstractPath, parent::AbstractPath)
```

Comment:
I think a more general way of writing this would be to use the `parent` or `parents` functions rather than relying on the `segments` field. Basically, just to minimize how many methods a new path type would need to overload if the structure changed significantly.

```
function isdescendent(child::AbstractPath, parent::AbstractPath)
    normpath(parent) in parents(normpath(child))
end
```

Reply:
I thought about this when adding the functions, but there would need to be recursive calls to `parents`. The existing implementation is effectively an optimization. Would you still prefer a `parents`-based implementation?

Reply:
Why would it need to be recursive? `parents` returns an array of all nodes in the path for exactly this kind of use case. If you'd like to see a more efficient implementation of `parents` then we can make that change in a separate PR.

Reply:
Honestly, the reason I didn't bother implementing these function earlier was because I figured these operations were simply enough to perform adhoc. I've since been convinced otherwise given that Julia's deprecation system is pretty flexible now.

Reply:
```julia
julia> using FilePathsBase
[ Info: Precompiling FilePathsBase [48062228-2e41-5def-b9a4-89aafe57970f]

julia> function isdescendent(child::AbstractPath, parent::AbstractPath)
           normpath(parent) in parents(normpath(child))
       end
isdescendent (generic function with 1 method)

julia> isdescendent(p"/a/b", p"/a")
true

julia> isdescendent(p"/a/b", p"/a/b")
false
```

Reply:
I just realized I missed a test, sorry about that! Does the commit I just pushed to this branch clarify anything?